daffodil-tdml-processor/src/test/scala/org/apache/daffodil/tdml/TestTDMLRunnerLogs.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.tdml
+
+import org.junit.Test
+import org.junit.AfterClass
+import org.apache.daffodil.Implicits._
+import org.junit.Assert.fail
+
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.{Logger => Log4jLogger}
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.core.config.Configurator
+
+object TestTDMLRunnerLogs {
+  val runner = Runner("/test/tdml/", "testLogs.tdml")
+  val tdmlAppender = TDMLAppender.createAppender
+
+  if(LogManager.getRootLogger().asInstanceOf[Log4jLogger].getAppenders().get("org.apache.tdml")==null){
+        val loggers = LogManager.getRootLogger().asInstanceOf[Log4jLogger]
+        loggers.addAppender(tdmlAppender)
+        tdmlAppender.start()
+        Configurator.setLevel("org.apache.daffodil", Level.INFO)
+    }
+
+  @AfterClass def shutDown(): Unit = {
+    runner.reset
+  }
+}
+
+class TestTDMLRunnerLogs {
+  import TestTDMLRunnerLogs._
+
+  @Test def test_logWhenExpectingSuccess() = { runner.runOneTest("logWhenExpectingSuccess", tdmlAppender) }

```

Comment:
Also, if we do as mentioned above, then we don't have to explicitly pass in something to capture logs. It's just something that is done by default by the TDML Processor and is always available to any TDML test.